.github/workflows/build-and-test-windows.yml

```diff
@@ -35,13 +56,22 @@ jobs:
         uses: actions/setup-go@v3
         with:
           go-version: 1.17
-      - name: Cache Go
+      - name: Cache Go Mod
+        id: go-mod-cache
         uses: actions/cache@v3
         with:
           path: |
             ~\go\pkg\mod
+          key: go-mod-cache-${{ runner.os }}-${{ hashFiles('**/go.sum') }}
+      - name: Install dependencies
+        if: steps.go-mod-cache.outputs.cache-hit != 'true'
+        run: make -j2 gomoddownload
+      - name: Cache Go Build
+        uses: actions/cache@v3
+        with:
+          path: |
             ~\AppData\Local\go-build
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          key: go-build-cache-${{ matrix.group }}-${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
```

Comment:
`setup-environment` is not per matrix correct? Why would that happen?

Reply:
Ah, I missed that `matrix.group` is included in the key.

We expect the build to be the same for all groups though, right?

Reply:
I think the "build" contains "built" artifacts which for a group will be the same over time, but not the same between groups, since I build/test a subset of the artifacts correct?

Reply:
I get it now. Been a while since I thought much about the artifacts `go test` produces.